src/pudl/transform/params/__init__.py

```diff
@@ -1,10 +1,22 @@
-"""A subpackage for defining data source specific transformation parameters.
+"""A subpackage for defining data source-specific transformation parameters.
 
 Each module in this subpackage is associated with and named after a single dataset (e.g.
-ferc1, eia923) and must define a dictionary named ``TRANSFORM_PARAMS`` in which the keys
-are all valid PUDL database table names provided by that dataset, and the values are
-are constants which can be used as parameters to construct a TableTransformParams class.
+``ferc1`` or ``eia923``) and must define a dictionary named ``TRANSFORM_PARAMS``.
+This dictionary is a nested data structure with 2 or 3 levels of keys:
 
-These dictionaries are used by the AbstractTableTransformer class to look up the
-parameters to be used in transforming a table, based on its name.
+* The first level has keys that table names (e.g. ``plants_steam_ferc1``).
+* The second level has keys that are the names of transform functions (e.g.
+  ``convert_units``).
+* In the case of transform functions that operate on a single column and implement the
+  :class:`pudl.transform.classes.ColumnTransformFunc` :class:`Protocol` the third level
+  of keys is the name of the column the transform should be applied to.
+
+The leaves of this tree structure are dictionaries of keyword arguments used to
+instantiate :class:`pudl.transform.classes.TransformParams` objects of the type that
+corresponds to the associated transform function, for use with the table or column
+that's been identified.
+
+These dictionaries are used by :class:`pudl.transform.classes.AbstractTableTransformer`
+to look up the parameters to be used in transforming a table based on the table name.
```

Comment:
this whole doc string is very good content and i fear it will not be seen. idrk what to do about that bc i can see why this is a good place for it. maybe in the `TableTransformParams` docs?

Reply:
As it is, it'll primarily show up in the documentation online at the top level of the params subpackage. I can sprinkle some references to that documentation page throughout the relevant places. There is more structure here than is encapsulated in the `TableTransformParams` classes -- a whole dictionary of `TableTransformParams` specifications for each of the different data sources (right now just FERC 1), which are then brought together to look up individual table transform parameters by `table_id`

Reply:
I've added a link from the `TableTransformParams` docstring.